qual/src/main/java/com/virtuslab/qual/guieffect/UIThreadUnsafe.java

```diff
@@ -5,9 +5,26 @@
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
-// TODO (typetools/checker-framework#3252): replace with proper @Heavyweight
+import org.checkerframework.checker.guieffect.qual.UIEffect;
 
-// Must have runtime retention to be visible to ArchUnit tests.
+/**
+ * Used to mark methods that should NOT be executed on UI thread
+ * (in case of IntelliJ Platform, on Swing's Event Dispatch Thread aka EDT)
+ * to avoid a UI freeze.
+ * These include methods that are heavyweight or blocking: typically, this means accessing disk and/or network.
+ * <p>
+ * Whether the annotation is used correctly is currently NOT enforced during compilation or runtime.
+ * TODO (typetools/checker-framework#3252): replace with proper {@code @Heavyweight} annotation.
+ * As for now, we use a ArchUnit test {@code com.virtuslab.archunit.UIThreadUnsafeMethodInvocationsTestSuite}
```

Comment:
```suggestion
 * As for now, we use an ArchUnit test {@code com.virtuslab.archunit.UIThreadUnsafeMethodInvocationsTestSuite}
```
should be "an" instead of "a" ?